rest/views.py

```diff
@@ -212,6 +213,43 @@ def get_queryset(self):
             slug=self.kwargs['vocab_slug']
         )
 
+    # pylint: disable=no-self-use
+    def remove_term_resource_links(self, vocab, new_types):
+        """
+        Remove Terms from LearningResources if the Term's Vocabulary's type
+        was removed.
+        """
+        old_types = set(t.name for t in vocab.learning_resource_types.all())
+        removed_types = old_types - new_types
+        for term in vocab.term_set.all():
+            for resource in term.learning_resources.all():
+                if resource.learning_resource_type.name in removed_types:
+                    term.learning_resources.remove(resource)
+
+    def update(self, request, *args, **kwargs):
+        """
+        Override to remove resource term links if resource type is
+        removed.
+        """
+        vocab = get_object_or_404(Vocabulary, slug=kwargs['vocab_slug'])
```

Comment:
if you fetch the vocabulary in this way here, why the `get_queryset` does it in a different way?

Wouldn't be better to get rid of `get_queryset`, override `get_object` and use `self.get_object` here?